src/keys/rsa.js

```diff
@@ -18,22 +18,34 @@ exports.generateKey = function (bits, callback) {
     return done(err)
   }
 
-  done(null, {
-    privateKey: pemToJwk(key.private),
-    publicKey: pemToJwk(key.public)
-  })
+  let res
+  try {
+    res = {
+      privateKey: pemToJwk(key.private),
+      publicKey: pemToJwk(key.public)
+    }
+  } catch (err) {
+    return done(err)
+  }
+
+  done(null, res)
 }
 
 // Takes a jwk key
 exports.unmarshalPrivateKey = function (key, callback) {
-  callback(null, {
-    privateKey: key,
-    publicKey: {
-      kty: key.kty,
-      n: key.n,
-      e: key.e
+  try {
+    key = {
+      privateKey: key,
+      publicKey: {
+        kty: key.kty,
+        n: key.n,
+        e: key.e
+      }
```

Comment:
Where would an exception be thrown in this operation?

Reply:
Never underestimate the power of undefined: `TypeError: Cannot read property 'e' of undefined`

Reply:
True, but in this case, it would never through that error, only if you did something like `key.e.something`. As long as if `key` exists and it is an object, `key.e` will never throw

Reply:
reverted this, but now inside a `setImmediate`.

Reply:
And what about `unmarshalPrivateKey(undefined, (err, res) => {})` ??

Reply:
@mkg20001 good point, fixing.